### DISCUSSION TITLE
CLI with export/import a collection

### DISCUSSION THREAD

**Discussion by kosssi** [directus/directus#11011] [Feature Requests]
It's cool if the CLI can export/import a collection.

I would like export collection schema from a Directus to import it to another Directus. In the documentation I see https://docs.directus.io/reference/cli/#migrate-schema-to-a-different-environment

I test with CLI like this:

```
cd directus.domain.fr
. .env
docker exec -it ${DIRECTUS_CONTAINER_NAME}_directus node ./node_modules/directus/cli.js schema snapshot ./snapshot.yml
docker cp ${DIRECTUS_CONTAINER_NAME}_directus:/directus/snapshot.yml ./snapshot.yml
mv ./snapshot.yml ../directus2.domain.fr
cd ../directus2.domain.fr
. .env
docker cp ./snapshot.yml ${DIRECTUS_CONTAINER_NAME}_directus:/directus/snapshot.yml
docker exec -it ${DIRECTUS_CONTAINER_NAME}_directus node ./node_modules/directus/cli.js schema apply ./snapshot.yml
```

But the CLI respond with a question "Would you like to continue?" with Delete collections and fields, I scared myself  😅 I would like append the collection ^^

I will test with the node SDK 😇

**Comment by rijkvanzanten** [comment]
> But the CLI respond with a question "Would you like to continue?" with Delete collections and fields, I scared myself 😅 I would like append the collection ^^

The schema snapshot system will update anything that needs to be changed in the receiving instance to match the snapshot, this includes removing collections that are no longer in use. It would be a nice addition to add a flag that will ignore the deletions in the schema apply 🙂

**Reply by kosssi** [reply]
It's same this discussion https://github.com/directus/directus/discussions/9214 ;)

**Reply by kosssi** [reply]
You know if a script js exist to export/import collection with js-sdk?

**Reply by oreilles** [reply]
There is no straightforward way to do it, but you could export all the table and fields information with
```ts
const collection = await sdk.collections.readOne('your_collection');
const fields = await sdk.fields.readMany('your_collection');
```
Then import them with: (untested)
```ts
await sdk.collections.createOne({ ...collection, fields })
```

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._

**Comment by jclaveau** [comment]
This extension https://github.com/ChappIO/directus-extension-seed allows to export all the data of the listed collections to a yaml file and, obviously, re-import them.

Ids will be exported also.


For roles and permissions this one works also well and has more features https://github.com/u12206050/directus-rbac-sync
